pkgs/development/compilers/llvm/common/default.nix

```diff
@@ -474,6 +474,28 @@ let
       bolt = callPackage ./bolt {
       };
     }
+    // lib.optionalAttrs (lib.versionAtLeast metadata.release_version "21") {
+      flang-unwrapped = callPackage ./flang {
+        # The C++ template instantiations in flang frequently cause g++ to be
+        # OOM-killed, so use clang++ instead.
+        stdenv = overrideCC stdenv buildLlvmTools.clang;
+      };
+      flang = wrapCCWith (
+        let
+          inherit (buildLlvmTools.clang) libcxx;
+        in
+        rec {
+          inherit libcxx;
```

Comment:
The `let in` and `rec` here doesn't look great. Is that really necessary?

Reply:
Eugh yeah I forgot to clean this bit up. Removed the `let in`.